### PR TITLE
Rule added to create events in  API groups for byohost

### LINKED ISSUE #596: 🐛 Host Agent is not able to add events
**What steps did you take and what happened:**
[A clear and concise description of what the bug is.]
`Host Agent` in the main does not have permissions to create events and the below warnings are seen in the logs. As it is not erroring out, the k8s node bootstrap is successful but the events are not present in `kubectl describe byohost host1`

`User "byoh:host:host1" cannot create resource "events" in API group "" in the namespace "default"`

**What did you expect to happen:**

The events should be created and printed with the `ByoHost` resource description.
``` 
  Normal  ByoHostAttachSucceeded     107s               byomachine-controller  Attached to ByoMachine test1-control-plane-tz2rx
  Normal  k8sComponentInstalled      88s (x4 over 35m)  hostagent-controller   Successfully Installed K8s components
  Normal  BootstrapK8sNodeSucceeded  69s (x4 over 34m)  hostagent-controller   k8s Node Bootstraped
```
**Anything else you would like to add:**
We need to add Role/Bindings for the host agent group i.e. `byoh:hosts` for the create permission of the events resource type.

### PR DESCRIPTION
**Pull request by shivi28**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #596 

**Additional information**

**Special notes for your reviewer**
Renamed one variable in VM templates for clusterclass

### CODE REVIEW

**Review by anusha94 — APPROVED** [review]
lgtm

File: test/e2e/data/infrastructure-provider-byoh/v1beta1/templates/vm/cluster-template-topology.yaml [hunk 1]
```diff
@@ -17,7 +17,7 @@ spec:
       - "192.168.0.0/16"
     serviceDomain: "cluster.local"
   topology:
-    class: ${CLUSTER_CLASS_NAME}
+    class: quickstart
```
Comment (dharmjit) [inline test/e2e/data/infrastructure-provider-byoh/v1beta1/templates/vm/cluster-template-topology.yaml:20]: Suggestion: if these changes are not related, it could have been a separate PR.

File: config/rbac/byohost_editor_role.yaml [hunk 2]
```diff
@@ -24,3 +24,14 @@ rules:
   - get
   - patch
   - update
+- apiGroups:
```
Comment (dharmjit) [inline config/rbac/byohost_editor_role.yaml:27]: Is it possible to narrow down the RBAC for events only for `ByoHost` CRDs?

File: test/e2e/data/infrastructure-provider-byoh/v1beta1/templates/vm/cluster-template-topology.yaml (hunk 1 above)
Comment (shivi28) [reply test/e2e/data/infrastructure-provider-byoh/v1beta1/templates/vm/cluster-template-topology.yaml:20]: Sure 👍

File: config/rbac/byohost_editor_role.yaml (hunk 2 above)
Comment (shivi28) [reply config/rbac/byohost_editor_role.yaml:27]: done

**Review by mayur-tolexo — APPROVED** [review]
LGTM

**Comment by shivi28** [comment]
lgtm
